client/src/components/Common/TextSummary.vue

```diff
@@ -0,0 +1,61 @@
+<script setup lang="ts">
+import { computed } from "vue";
+import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
+import { library } from "@fortawesome/fontawesome-svg-core";
+import { faChevronUp, faChevronDown } from "@fortawesome/free-solid-svg-icons";
+
+const props = defineProps({
+    description: {
+        type: String,
+        required: true,
+    },
+    showDetails: {
+        type: Boolean,
+        default: false,
+    },
+});
+
+const emit = defineEmits<{
+    (e: "update:show-details", showDetails: boolean): void;
+}>();
+
+const propShowDetails = computed({
+    get: () => {
+        return props.showDetails;
+    },
+    set: (val) => {
+        emit("update:show-details", val);
+    },
+});
+
+//@ts-ignore bad library types
+library.add(faChevronUp, faChevronDown);
+const collapsedEnableIcon = "fas fa-chevron-down";
+const collapsedDisableIcon = "fas fa-chevron-up";
+
+// max length (can be a prop, maybe?)
+const n = 150;
```

Comment:
Sure, a prop with 150 char default sounds like a good approach.

Reply:
Alrighty, I'll turn it into a prop, thanks!